table/server/underdb/glue/src/main/java/alluxio/table/under/glue/GlueDatabase.java

```diff
@@ -377,15 +378,26 @@ public UdbTable getTable(String tableName) throws IOException {
 
   private List<Partition> batchGetPartitions(AWSGlueAsync glueClient, String tableName)
       throws IOException {
+    // TODO(shouwei): make getPartition multi-thread to accelerate the large table fetching
     List<Partition> partitions = new ArrayList<>();
+    String nextToken = null;
     try {
-      GetPartitionsRequest getPartitionsRequest =
-          new GetPartitionsRequest()
-              .withCatalogId(mGlueConfiguration.get(Property.CATALOG_ID))
-              .withDatabaseName(mGlueDbName)
-              .withTableName(tableName);
-      if (glueClient.getPartitions(getPartitionsRequest).getPartitions() != null) {
-        partitions = glueClient.getPartitions(getPartitionsRequest).getPartitions();
+      do {
+        GetPartitionsRequest getPartitionsRequest =
+            new GetPartitionsRequest()
+                .withCatalogId(mGlueConfiguration.get(Property.CATALOG_ID))
+                .withDatabaseName(mGlueDbName)
+                .withTableName(tableName)
+                .withMaxResults(mGlueConfiguration.getInt(Property.MAX_GLUE_FETCH_PARTITIONS))
+                .withNextToken(nextToken);
+        GetPartitionsResult getPartitionsResult = glueClient.getPartitions(getPartitionsRequest);
+        getPartitionsResult.getPartitions().forEach(partition -> partitions.add(partition));
+        nextToken = getPartitionsResult.getNextToken();
+        LOG.debug("Glue table {}.{} added {} partitions.",
+            mGlueDbName, tableName, partitions.size());
```

Comment:
should this be `getPartitionsResult.getPartitions().size()` instead of `partitions.size()`?

Reply:
It allows us to track the progress of `getPartitions`. Is it better to show the total added partitions for the current table?

Reply:
I think both can be useful. How about something like "Glue table {}.{} adding {} batch partitions with {} total partitions."

Reply:
yeah, it makes sense, I can add both of them.